### PR TITLE
Assorted minor documentation changes

### PR DESCRIPTION
**Pull request by melissachang**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -380,29 +385,25 @@ To add a new supported tool:
...
+      If you need additional environment variables, you can pass them into `CommandRunner.runToolCommand()`.
+   8. You can mount directories on the host machine to the Docker container by populating a second
+         `Map` and passing it to the same `DockerAppsRunner.runToolCommand` method. The current working directory is always
+         mounted to the Docker container.
    9. Publish the new Docker image and update the default image that the CLI uses to the new version
-   (see instructions in section above).
+         (see instructions in section above).
 
 #### Commands
 The `command` package contains the hierarchy of the commands as they appear to the user.
```
Comment (jaycarlton) [inline CONTRIBUTING.md:409]: Below on line 563, it says "Tests only run in docker mode." That's partially true today, and will become less so soon. I'd document that when we setup tests we pull the docker container, and Docker is present in the test environment, so tests are installed in Docker mode (if that's not more confusing). They're not necessarily _run_ in Docker mode, as some of them switch `app-launch` to `LOCAL_PROCESS`.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -592,16 +595,22 @@ See PF-1395 for switching this repo to 3.)
...
 
 </td>
 <td valign="top">
 
-**Pet ADC**
+**Test user ADC**
+
+*Nextflow*
+
+Populate `.config/gcloud/application_default_credentials.json`.
```
Comment (jaycarlton) [inline CONTRIBUTING.md:612]: What populates this file?

By the way, are there any responsibilities on test writers or maintainers to clean up the auth at the end of the test run?

File: README.md [hunk 3]
```diff
@@ -393,6 +419,12 @@ Terra groups are managed by SAM. These commands are utility wrappers around the
 
 The `enterprise-pilot-testers` group is used for managing access to the default WSM spend profile.
```
Comment (jaycarlton) [inline README.md:420]: This is an operational detail that belongs in a private repo or document, not in the open source one.

File: docker/Dockerfile [hunk 4]
```diff
@@ -25,7 +25,7 @@ RUN curl -O https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cl
     gcloud config set component_manager/disable_update_check true
 
 # install Nextflow
-ENV NXF_VER 21.05.0-edge
+ENV NXF_VER 21.10.6
```
Comment (jaycarlton) [inline docker/Dockerfile:28]: is this change related to this PR?

File: src/main/java/bio/terra/cli/app/DockerCommandRunner.java [hunk 5]
```diff
@@ -92,20 +86,10 @@ protected int runToolCommandImpl(String command, Map<String, String> envVars)
...
-
-          // set the env var to point to the file, since it's not in the default gcloud location
-          envVars.put("GOOGLE_APPLICATION_CREDENTIALS", adcFileOnContainer.toString());
-        }
+      if (adcCredentialsFile.isPresent()
+          && adcCredentialsFile.get().equals(AppDefaultCredentialUtils.getDefaultGcloudADCFile())) {
+        logger.info(
+            "ADC backing file is in the default location and is already mounted in the gcloud config directory");
       } else {
         logger.info("ADC set by metadata server.");
```
Comment (jaycarlton) [inline src/main/java/bio/terra/cli/app/DockerCommandRunner.java:94]: Maybe indicate that this is the VM's local metadata server, or however you say it.

File: src/main/java/bio/terra/cli/app/LocalProcessCommandRunner.java [hunk 6]
```diff
@@ -66,13 +66,9 @@ protected int runToolCommandImpl(String command, Map<String, String> envVars)
...
-      if (adcCredentialsFile.isPresent()) {
-        if (adcCredentialsFile.get().equals(AppDefaultCredentialUtils.getDefaultGcloudADCFile())) {
-          logger.info("ADC backing file is in the default location");
-        } else {
-          // set the env var to point to the file, since it's not in the default gcloud location
-          envVars.put("GOOGLE_APPLICATION_CREDENTIALS", adcCredentialsFile.toString());
-        }
+      if (adcCredentialsFile.isPresent()
+          && adcCredentialsFile.get().equals(AppDefaultCredentialUtils.getDefaultGcloudADCFile())) {
+        logger.info("ADC backing file is in the default location");
```
Comment (jaycarlton) [inline src/main/java/bio/terra/cli/app/LocalProcessCommandRunner.java:71]: How do we know it's set by the metadata server in the case where it's present but doesn't equal the default?

File: src/main/java/bio/terra/cli/app/utils/AppDefaultCredentialUtils.java [hunk 7]
```diff
@@ -95,7 +83,9 @@ public static GoogleCredentials getApplicationDefaultCredentials() {
     try {
       return GoogleCredentials.getApplicationDefault();
     } catch (IOException ioEx) {
-      throw new UserActionableException("Application default credentials are not defined.", ioEx);
+      throw new UserActionableException(
+          "Application default credentials are not defined. Run `gcloud auth application-default login`",
```
Comment (jaycarlton) [inline src/main/java/bio/terra/cli/app/utils/AppDefaultCredentialUtils.java:87]: That advice works for local mode, but what if we're on a notebook and somehow can't get the ADCs?

File: src/main/java/bio/terra/cli/businessobject/User.java [hunk 8]
```diff
@@ -229,45 +210,6 @@ public void fetchPetSaCredentials() {
     logger.debug("Enabled pet SA impersonation");
   }
```
Comment (jaycarlton) [inline src/main/java/bio/terra/cli/businessobject/User.java:212]: I thought you already deleted this in another PR.

**Review by jaycarlton — APPROVED** [review]
Thanks for diving into this! This is important information that was very fuzzy for me before and is less so now. You are now the official expert on CLI auth.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -380,29 +385,25 @@ To add a new supported tool:
...
-   Copy/paste an existing class in that same package as a starting point.
-   5. Add it to the list of tools shown by `terra app list` by adding the new command class to
-   the list of sub-commands in the `@Command` annotation of the `Main.class`. This means you can
-   invoke the command by prefixing it with terra (e.g. `terra dsub -version`).
+      (e.g. `terra app execute dsub --version`). This command just runs the Docker container and executes the command,
+      without requiring any new Java code. This `terra app execute` command is intended for debugging only; this won't
+      be how users call the tool.
+   4. Add a new command class in the `src/main/java/bio/terra/cli/command/app/passthrough` package. Copy/paste an
+      existing class in that same package as a starting point.
+   5. Add it to the list of tools shown by `terra app list` by adding the new command class to the list of sub-commands
```
[the remark this answers is not in the document]
Comment (yuhuyoyo) [reply CONTRIBUTING.md:393]: i saw `git` when I run `terra app list`. It's also defined here https://github.com/DataBiosphere/terra-cli/blob/main/src/main/java/bio/terra/cli/command/app/List.java#L55.

[the remark this answers is not in the document]
Comment (melissachang) [reply CONTRIBUTING.md]: Hmm, I'm not sure what `setStopAtPositional()` does. We're not setting it for git, but I tried a bunch of `terra git` commands and they all worked...

File: CONTRIBUTING.md (hunk 1 above)
Comment (melissachang) [reply CONTRIBUTING.md]: Changed to `Tests run in docker mode by default` and linked to where I think that is set.

(I don't think `‎‎tools/install.sh` runs for unit tests. Unit tests run the Java commands directly.)

File: CONTRIBUTING.md (hunk 2 above)
Comment (melissachang) [reply CONTRIBUTING.md]: If you search repo for `config/gcloud/application_default_credentials.json`, you'll find it. I thought about linking to the code directly, but there's a bunch of link I should create to be consistent. I think it's easy enough to search the repo.

[Tests delete `.gcloud/config` before they run.](https://github.com/DataBiosphere/terra-cli/blob/79a938e56f2d95111aeec54175b6d38d9e5deb79/src/test/java/harness/baseclasses/ClearContextUnit.java#L24)

File: README.md (hunk 3 above)
Comment (melissachang) [reply README.md]: Done

File: docker/Dockerfile (hunk 4 above)
Comment (melissachang) [reply docker/Dockerfile]: This is gone now that I rebased

File: src/main/java/bio/terra/cli/app/utils/AppDefaultCredentialUtils.java (hunk 7 above)
Comment (melissachang) [reply src/main/java/bio/terra/cli/app/utils/AppDefaultCredentialUtils.java:87]: GCP Notebook should always have ADC.

File: src/main/java/bio/terra/cli/businessobject/User.java (hunk 8 above)
Comment (melissachang) [reply src/main/java/bio/terra/cli/businessobject/User.java]: Gone after rebase

**Comment by melissachang** [comment]
@yuhuyoyo , can you review since Chris is OOO?

**Review by yuhuyoyo — APPROVED** [review]
